mission/BouysTasksNjord/bouys_tasks.py

```diff
@@ -0,0 +1,338 @@
+#!/usr/bin/python3
+#Written by Sigurd von Brandis, Student
+
+import rospy
+import smach
+import math
+from vortex_msgs.srv import Waypoint, WaypointRequest, WaypointResponse
+from nav_msgs.msg import Odometry
+from std_msgs.msg import Float64
+from tf.transformations import euler_from_quaternion
+from vortex_msgs.msg import DetectedObjectArray, DetectedObject
+from update_objects_data import DetectedObjectsData, UpdateDataNode
+
+
+class Idle(smach.State):
+    # Need to be added that idle make the vessel stop moving, before entering search state.
+    def __init__(self):
+        smach.State.__init__(
+            self,
+            outcomes=['decideNextState', 'search'],  #, 'stop'
+            input_keys=['closest_object', 'object_search_attempts'],
+            output_keys=['object_search_attempts'])
+
+    def execute(self, userdata):
+        rospy.loginfo('Executing Idle')
+
+        if userdata['closest_object'][1] == '':
+            #userdata['object_search_attempts'] += 1
+            if userdata['object_search_attempts'] >= 5:
+                return 'stop'
+            else:
+                return 'search'
+        else:
+            userdata['object_search_attempts'] = 0
+            return 'decideNextState'
+
+
+class Search(smach.State):
+    # Must be changed so it returns object search attempts. Should be done here, not in idle state.
+    def __init__(self):
+        smach.State.__init__(self, outcomes=['idle'])
+        self.task = "Buoy"
+        self.odom = Odometry()
+        self.rate = rospy.Rate(10)
+
+        self.heading_pub = rospy.Publisher(
+            "/guidance_interface/desired_heading", Float64, queue_size=1)
+
+        rospy.Subscriber("/odometry/filtered", Odometry, self.odom_cb)
+
+    def odom_cb(self, msg):
+        self.odom = msg
+
+    def within_acceptance_margins(setpoint, current):
+        error = abs(setpoint - current)
+        if error < 0.1:
+            return True
+        return False
+
+    def yaw_to_angle(self, angle):
```

Comment:
what does yaw_to_angle mean? Rename to better reflect the functionality of the function, and comment the code